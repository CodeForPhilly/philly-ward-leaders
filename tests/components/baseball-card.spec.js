import { mount,shallowMount,RouterLinkStub, renderToString  } from '@vue/test-utils'
//import router from '../../src/router'

import BaseballCard from '../../src/components/baseball-card/index.vue'
import { createSSRApp } from 'vue'

describe('Baseball card', () => {
  it('Matches snapshot', async () => {
    const propsData = {
      name: 'John Doe',
      ward: 1,
      party: 'democratic',
      photoUrl: 'https://example.com/image.png',
      registeredVotersParty: 1000,
      turnoutParty: 400,
      turnoutTotal: 600,
      divisionCount: 30,
      committeePersonCount: 25
    }
    const htmL_text = await renderToString(BaseballCard, {
      propsData,
      stubs: {
        'router-link': {
          template: '<a><slot /></a>'
        }
      }
    })
    console.log(htmL_text)
    expect(htmL_text).toMatchSnapshot()
  })

  it('Includes subward in URL', () => {
    const propsData = {
      name: 'John Doe',
      ward: 1,
      subWard: 'A',
      party: 'democratic',
      photoUrl: 'https://example.com/image.png',
      registeredVotersParty: 1000,
      turnoutParty: 400,
      turnoutTotal: 600,
      divisionCount: 30,
      committeePersonCount: 25
    }
    const wrapper = mount(BaseballCard, {
      propsData,
     })
    const url = wrapper.vm.url
    expect(url).toBe('/leaders/democratic/1A/john-doe')
  })
})
