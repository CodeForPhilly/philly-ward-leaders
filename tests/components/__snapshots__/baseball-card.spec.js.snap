// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Baseball card Matches snapshot 1`] = `
<div class="column is-one-third flip-container">
    <div class="flipper">
        <div class="front"><svg></svg></div>
        <div class="back"><svg></svg></div> <svg class="contents">
            <foreignObject width="100%" height="100%">

                <body>
                    <div class="stats">
                        <h3>
                            John Doe
                            <span class="ward-ordinal">
                                1st Ward

                            </span></h3>
                        <dl><dt>Voters (D)</dt>
                            <dd>1,000</dd> <dt>Turnout (D)</dt>
                            <dd>40%</dd> <dt>Vacancies</dt>
                            <dd>35</dd>
                        </dl>
                        <!---->
                    </div>
                </body>
            </foreignObject>
        </svg>
    </div>
</div>
`;
